ssz/src/main/java/tech/pegasys/artemis/ssz/SSZTypes/Bitvector.java

```diff
@@ -40,6 +42,26 @@ public Bitvector(Bitvector bitvector) {
     this.size = bitvector.size;
   }
 
+  public Bitvector(List<Integer> indicesToSet, int size) {
+    this.size = size;
+    this.data = new BitSet(size);
+    if (indicesToSet.size() == 0) {
+      return;
+    }
+    checkArgument(
+        indicesToSet.stream().distinct().count() != indicesToSet.size(),
+        "Bitvector: Bit indices to set are not distinct");
```

Comment:
Instead of adding a check here, you could just accept a `Set` of indices:
```
public Bitvector(Set<Integer> indicesToSet, int size) {
```

Reply:
That's quite an expensive check to add as well and I'm not sure it actually matters.  It's like creating a `Set` from a `List` - duplicates should just be ignored.